opts/mount.go

```diff
@@ -112,6 +113,33 @@ func (m *MountOpt) Set(value string) error {
 			if err != nil {
 				return fmt.Errorf("invalid value for %s: %s", key, val)
 			}
+			logrus.Warn("bind-nonrecursive is deprecated, use bind-recursive=disabled instead")
+		case "bind-recursive":
+			valS := val
+			// Allow boolean as an alias to "enabled" or "disabled"
+			if b, err := strconv.ParseBool(valS); err == nil {
+				if b {
+					valS = "enabled"
+				} else {
+					valS = "disabled"
+				}
+			}
+			switch valS {
+			case "enabled": // read-only mounts are recursively read-only if Engine >= v25 && kernel >= v5.12
+				// NOP
+			case "disabled": // alias of bind-nonrecursive=true
+				bindOptions().NonRecursive = true
+			case "readonly-classic": // conforms to Docker v24; read-only mounts are recursively mounted but not recursively read-only
```

Comment:
`readonly-classic` doesn't feel right to me, as we're trying to split the "recursion" from the "readonlyness" with this.

How about `writable`, which makes it clear that recursive mounts will be writable/that was the legacy behavior?

---

I also think setting `ReadOnly` here is a smell -- we should instead have some check that errors when `ReadOnly` has not been set by the user (e.g. `readonly=false,bind-recursive=readonly` should be an error in validation).

Reply:
What about "classic" or "legacy"?

Reply:
I still think `classic` or `legacy` don't really describe what it **does**, and when documenting this option and the default for old engine versions, it really hammers home the old behavior. I don't object to an alias under those names, but I would prefer avoiding it as I think they're meaningless unless you have an excellent understanding of the kernel/engine history around this.